### ISSUE TITLE
ideas shared on #1450

### ISSUE THREAD

**Issue by niphlod** [web2py/web2py#1467]
let's discuss here the move of caching models, controllers and views (both compiled or not).
@ilvalle ran some benchmarks that should be replicated by the most of us if we want to "consolidate" the change.
Also,  @mdipierro commented

> The test function allows to run doctests from admin. Wan't this break backward compatibility for compiled apps? I am on with it but we should make a very prominent note of it.

**Comment by ilvalle** [comment]
Right, benchmarks can be easily replicated using apache benchmark as shown in https://github.com/web2py/web2py/pull/1450. The best would be to run test on a normal hd (no ssd).

**Comment by ilvalle** [comment]
After https://github.com/web2py/web2py/commit/b19c3419ec9313ce9cac62849e7d6f247ec40f65 a compiled view is neither cached nor executed. In
https://github.com/web2py/web2py/blob/master/gluon/compileapp.py#L698 the compiled files is actually read, but in https://github.com/web2py/web2py/blob/master/gluon/compileapp.py#L710 it is the 'non compiled' that is read and executed.
